### PR TITLE
Check vector length in cross()

### PR DESCRIPTION
**Pull request by garrison**
`cross` conceptually assumes the vectors are each of length three, but previously this has not been checked.  (This code is essentially unchanged since it was introduced in e673b1c8061b6486589d299cf5deff41b76f1fc6.)  This is technically breaking, but I can't imagine anybody should have been passing vectors of length > 3 here, so I think it's safe to merge without deprecation warning.

### CODE REVIEW

**Comment by garrison** [comment]
While we're at it, I'll also note that `StaticArrays.jl` has [an additional version of `cross`](https://github.com/JuliaArrays/StaticArrays.jl/blob/b454bc9131333da153f49182c7829c385fb2a6f5/src/linalg.jl#L210-L212) that takes two 2-vectors and returns the z component of the cross product of those vectors in the x-y plane.  It was added by @SimonDanisch in JuliaArrays/StaticArrays.jl@c5880408cc9e15fe3b266738abfcdf4d1e357a0b, without any discussion (to the best of my knowledge).  This can, of course, be added here either now or later (e.g. post-1.0) if people so desire.

File: stdlib/LinearAlgebra/src/generic.jl [hunk 1]
```diff
@@ -109,8 +109,12 @@ julia> cross(a,b)
  0
 ```
 """
-cross(a::AbstractVector, b::AbstractVector) =
+function cross(a::AbstractVector, b::AbstractVector)
+    if !(length(a) == length(b) == 3)
+        throw(DimensionMismatch("cross product is only defined for vectors of length 3"))
+    end
     [a[2]*b[3]-a[3]*b[2], a[3]*b[1]-a[1]*b[3], a[1]*b[2]-a[2]*b[1]]
```
Comment (alyst) [inline stdlib/LinearAlgebra/src/generic.jl]: Since the length is checked, this could be annotated with `@inbounds`.

Comment (GregPlowman) [reply stdlib/LinearAlgebra/src/generic.jl]: Offset vectors?

Comment (garrison) [reply stdlib/LinearAlgebra/src/generic.jl]: @GregPlowman what about them?  Last I checked, `length` is not implemented for offset vectors, so this function would simply fail for them (assuming nothing has changed).

Comment (GregPlowman) [reply stdlib/LinearAlgebra/src/generic.jl]: OK, nothing to worry about then.
Sorry for the distraction.

Comment (alyst) [reply stdlib/LinearAlgebra/src/generic.jl]: To be on the safe side, you can add `firstindex(a) == firstindex(b) == 1` check.
I suppose for `Vector` it should be optimized into `true` at compile time.

**Comment by garrison** [comment]
This is ready to merge, assuming somebody can clarify that 1,2,3 will always be valid indices for an `AbstractVector` with `length(v) == 3` (or in other words, that `length()` will always fail for offset arrays).

**Comment by mbauman** [comment]
That's currently the case, but there is a desire to change that in the future.  Maybe just unpack the arrays via iteration?  `a1, a2, a3 = a`.  I'd wager that LLVM isn't caching the loads, so that has the advantage of being slightly faster, too.

**Review by mbauman — APPROVED** [review]

**Pull request merged**